app/scripts/background.js

```diff
@@ -331,6 +332,34 @@ function maybeDetectPhishing(theController) {
   );
 }
 
+/**
+ * Overrides the Content-Security-Policy Header, acting as a workaround for an MV2 Firefox Bug.
+ */
+function overrideContentSecurityPolicyHeader() {
+  browser.webRequest.onHeadersReceived.addListener(
+    ({ responseHeaders }) => {
+      for (const header of responseHeaders) {
+        if (header.name.toLowerCase() === 'content-security-policy') {
+          const contentSecurityPolicy = CSP.parse(header.value);
+          const nonce = `'nonce-${btoa(browser.runtime.getURL('/'))}'`;
+          const scriptSrc = Object.keys(contentSecurityPolicy).find(
+            (directive) => directive.toLowerCase() === 'script-src',
+          );
+          if (scriptSrc) {
+            contentSecurityPolicy[scriptSrc].push(nonce);
+          } else {
+            contentSecurityPolicy['script-src'] = [nonce];
+          }
```

Comment:
_Ideally_ we'd only ever modify the CSP if we absolutely must. I think trying to do that will be very difficult to do correctly/perfectly, so maybe we can limit the checks to cases where there is already a `script-src` or a `default-src` directive present?

That said, it is likely that 99% of all CSP headers already contain a `default-src` directive, so maybe adding checks here will be unnecessary. I'll let you decide.